### PR TITLE
Makefile.inc: add compile-debug target

### PR DESCRIPTION
**Pull request by eriknyquist**
"make compile" now compiles without debug symbols. "make compile-debug"
compiles with debug symbols, and also omits the "-Os" optimisation flag

If testing with CODK-M, you must apply 01org/CODK-M#33 to your CODK-M repo.
If testing with CODK-A, you must apply 01org/CODK-A#24 to your CODK-A repo.

@calvinatintel @SidLeung please review

### CODE REVIEW

**Comment by calvinatintel** [comment]
Can this be tested on its own, or does it need to be combined with some other PR?

**Review by calvinatintel — COMMENTED** [review]
Line 172, clean target might need to be tightened a little bit

**Comment by calvinatintel** [comment]
Do you know why we have `clean` and `clean-all`? Shouldn't `clean` remove everything?

**Comment by eriknyquist** [comment]
@calvinatintel yeah, you're right. I don't even see the "clean" target being used anywhere, CODK-M and CODK-A both invoke 'clean-all'. Let's do it in a different PR, since it's a non-functional change.

File: Makefile.inc [hunk 1]
```diff
@@ -119,12 +119,18 @@ $(OUTNAME).elf: $(OBJS) $(LIBOBJS) $(CORE_A)
...
+
+compile: CFLAGS+=-Os
+compile: CPPFLAGS+=-Os
+compile: do-compile
+
+compile-debug: CFLAGS+=-g
+compile-debug: CPPFLAGS+=-g
+compile-debug: do-compile
 
 upload: compile
```
Comment (calvinatintel) [inline Makefile.inc]: `upload` target currently compiles before uploading. This was put in place to mimic how IDE re-compiles upon pressing upload. We should take it out.

**Review by calvinatintel — APPROVED** [review]

**Pull request merged**